### ISSUE TITLE
getTags returns only tags first time

### ISSUE THREAD

**Issue by jzeferino** [OneSignal/OneSignal-Android-SDK#97]
After calling `getTags` for the first time it always returns null.

The problem is in this lines:

https://github.com/OneSignal/OneSignal-Android-SDK/blob/master/OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java#L777

`            if (tags.serverSuccess) getTagsCall = true;
`

and after settings this flag to true it never goes to false and in the next calls it will never call the getTags:

https://github.com/OneSignal/OneSignal-Android-SDK/blob/master/OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java#L776
 `           final OneSignalStateSynchronizer.GetTagsResult tags = OneSignalStateSynchronizer.getTags(!getTagsCall);`

https://github.com/OneSignal/OneSignal-Android-SDK/blob/master/OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignalStateSynchronizer.java#L698
```
static GetTagsResult getTags(boolean fromServer) {
      lastGetTagsResponse = null;

      if (fromServer) {
      // Never enter here after the first time.
```

**Comment by jkasten2** [comment]
@jzeferino Thanks for the report. Are you calling `sendTag(s)` in your code as well?

From a quick review of the SDK code it seems there could be an issue where calling `getTags` could overwrite `sendTag(s)` values if they have not been synced to the OneSignal server yet.

**Comment by jzeferino** [comment]
@jkasten2 your welcome. Yes I'm calling` sendTag(s)` too, but only after a `getTags` call.
Do you plan to solve this issue?

**Comment by jkasten2** [comment]
@jzeferino I haven't been able to reproduce an issue where the first call to `getTags` would return `null` if there are tags already set on the user server side or client side. We already have a unit test around this behavior as well.
https://github.com/OneSignal/OneSignal-Android-SDK/blob/d978cce51289b0f2be76d46c03943c4f9e0cdc87/OneSignalSDK/app/src/test/java/com/test/onesignal/MainOneSignalClassRunner.java#L1066

Could you list the steps to reproduce the issue in detail?

**Comment by jzeferino** [comment]
Maybe my first post wasn't clear.

> After calling getTags for the first time it always returns null.

To reproduce this you should call the `getTags` **AFTER** the first time.
By others words, the 2nd, 3rd, etc... call to `getTags` always return null due to the lines i mentioned above. Please let me know if you can reproduce it.

**Comment by jkasten2** [comment]
@jzeferino Sorry I misread that, I have fixed the issue in commit c4012fefb4fc1c8a7b34b1017d8ec7a91ca7b832. There is now additional tests around this method call to prevent any recurrence of an issue like this.

**Comment by jzeferino** [comment]
@jkasten2 thank you
